lib/solr_wrapper/instance.rb

```diff
@@ -204,55 +240,51 @@ def validate!(file)
 
     ##
     # Run a bin/solr command
+    # @param [String] cmd command to run
+    # @param [Hash] options key-value pairs to transform into command line arguments
+    # @param [Array] boolean_flags to include in command (ie. "-c")
+    # @return [StringIO] an IO object for the executed shell command
     # @see https://github.com/apache/lucene-solr/blob/trunk/solr/bin/solr
-    def exec(cmd, options = {})
-      output = !!options.delete(:output)
-      args = [solr_binary, cmd] + solr_options.merge(options).map { |k, v| ["-#{k}", "#{v}"] }.flatten
-
+    # @example start solr in cloud mode on port 8983
+    #   exec('start', {p: '8983'}, ["-c"])
+    def exec(cmd, options = {}, boolean_flags = [])
+      boolean_flags ||= []
+      args = [solr_binary, cmd] + boolean_flags + solr_options.merge(options).map { |k, v| ["-#{k}", "#{v}"] }.flatten
       if IO.respond_to? :popen4
         # JRuby
         env_str = env.map { |k, v| "#{Shellwords.escape(k)}=#{Shellwords.escape(v)}" }.join(" ")
         pid, input, output, error = IO.popen4(env_str + " " + args.join(" "))
         @pid = pid
         stringio = StringIO.new
-        if verbose? && !output
+        IO.copy_stream(output, stringio)
```

Comment:
It wasn't actually doing anything meaningful other than unnecessarily preventing you from having access to the output in some cases.  Also, you need to at least capture the output within the method call in order return meaningful error messages.  Once you have the output in hand, there's not much point in _not_ returning it.